### PR TITLE
Feature/event reporter

### PR DESCRIPTION
**Pull request by KieranDaviesV**
### What

Added the event reporter

### How to review

Tested with fake data

### Who can review

Anyone

### CODE REVIEW

File: cmd/dp-observation-extractor/main.go [hunk 1]
```diff
@@ -22,13 +24,13 @@ func main() {
 		os.Exit(1)
 	}
 	log.Debug("loaded config", log.Data{"config": config})
-
-	s3, err := s3.New(config.AWSRegion)
+	//TODO Delete once finished testing.
```
Comment (eldeal) [inline cmd/dp-observation-extractor/main.go]: Remove before merge

File: event/csv_handler.go [hunk 2]
```diff
@@ -38,11 +42,13 @@ func (handler CSVHandler) Handle(event *DimensionsInserted) error {
 	log.Debug("getting file", log.Data{"url": url, "event": event})
 	readCloser, err := handler.fileGetter.Get(url)
 	if err != nil {
+		handler.errorHandler.Handle(event.InstanceID, err)
```
Comment (eldeal) [inline event/csv_handler.go]: There's a lot of stutter in this function call. Perhaps the field on CSVHandler should just be `errors` so this call becomes handler.errors.Handle?

File: event/csv_handler_test.go [hunk 3]
```diff
@@ -90,7 +91,7 @@ func TestCsvHandler(t *testing.T) {
 		fileGetterStub := &eventtest.FileGetter{Reader: stubReadCloser}
 		observationWriterStub := &eventtest.ObservationWriter{}
 
-		eventHandler := event.NewCSVHandler(fileGetterStub, observationWriterStub)
+		eventHandler := event.NewCSVHandler(fileGetterStub, observationWriterStub, nil)
```
Comment (eldeal) [inline event/csv_handler_test.go]: Should there be some test cases added where we provide an error handler? Or is that all covered by the tests in the other package?

Comment (KieranDaviesV) [reply event/csv_handler_test.go]: I really struggled with this bit, I have tested it within the go-ns package I don't know if that's sufficient  enough tho

**Pull request closed without merging**